src/main/java/de/novatec/domainprimitives/object/ComposedValueObject.java

```diff
@@ -0,0 +1,68 @@
+package de.novatec.domainprimitives.object;
+
+import de.novatec.domainprimitives.validation.InvariantException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.lang.String.format;
+
+public abstract class ComposedValueObject {
+
+    private final List<String> violations;
+
+    protected ComposedValueObject() {
+        this.violations = new ArrayList<>();
+    }
+
+    /**
+     * Providing a starting point this stub could be used check on invariant violations.
+     */
+    protected abstract void validate();
+
+    /**
+     * Helping validation method to ensure the field is not null
+     *
+     * @param field The object with should be null-checked
+     * @param label The lable or a description of the field
+     */
+    protected void validateNotNull(Object field, String label) {
```

Comment:
Same here. I think we should move the concern of constraints out of the base object, where it is potentially unused.

Reply:
#2